uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request-ctrl.js

```diff
@@ -656,13 +670,14 @@ define([
             initContact.call(self);
 
             if (self.isMode('edit')) {
-              initialLeaveRequestAttributes = self.request.attributes();
+              initialLeaveRequestAttributes = angular.copy(self.request.attributes());
+              // fileUploader property deleted because it will not be used
+              // in object comparison
+              delete initialLeaveRequestAttributes.fileUploader;
```

Comment:
you are duplicating logic here. keep this plain
```js
initialLeaveRequestAttributes = angular.copy(self.request.attributes());
```
and do the removal of properties you don't need to compare in the `hasRequestChanged` function